src/tree.c

```diff
@@ -797,14 +803,33 @@ int git_treebuilder_remove(git_treebuilder *bld, const char *filename)
 
 int git_treebuilder_write(git_oid *oid, git_treebuilder *bld)
 {
+	int error;
+	size_t entrycount;
+	git_buf buffer = GIT_BUF_INIT;
+
+	entrycount = git_strmap_num_entries(bld->map);
+
+	/* Grow the buffer beforehand to an estimated size */
+	error = git_buf_grow(&buffer, entrycount * 72);
```

Comment:
Why is the caller growing the buffer instead of leaving the decision to the consumer?